### ISSUE TITLE
Out-of-memory when reading a HTTP response with large Content-Length

### ISSUE THREAD

**Issue by serhiy-storchaka** [python/cpython#119451]
`http.client.HTTPResponse.read()` (without argument) consumes the amount of memory specified by the value of the `Content-Lenght` header in the response before starting receiving the data from the socket. Normally, if the server does not send enough data, the client gets `IncompleteRead` error, but if `Content-Lenght` is too large, it can consume a large amount of memory and CPU time and cause swapping. Therefore, a maleficent server can cause a DOS attack on client by sending a small response.

### Linked PRs
* gh-119454
* gh-142138
* gh-142139
* gh-142140
* gh-142141
* gh-142142

**Comment by z00z00z00** [comment]
The CVE assigned is CVE-2025-13836

**Comment by serhiy-storchaka** [comment]
You are right, but the `_safe_read()` code was changed not in 152f0b8beea12e6282d284100b600771b968927a (bpo-41002), but in d6bf6f2d0c83f0c64ce86e7b9340278627798090 (bpo-36050). This is 3.8.0a4.

**Comment by frenzymadness** [comment]
Hmm, it seems to be more complex. AFAICT, 3.6 is not vulnerable, 3.8 is vulnerable, 3.9 is not, and 3.10+ are vulnerable again.
